packages/rocketchat-file-upload/lib/FileUpload.js

```diff
@@ -10,45 +10,43 @@ FileUpload = {
 		if (!Match.test(file.rid, String)) {
 			return false;
 		}
-
-		const user = Meteor.user();
+		const user = (file.userId) ? Meteor.user() : null;
 		const room = RocketChat.models.Rooms.findOneById(file.rid);
 		const directMessageAllow = RocketChat.settings.get('FileUpload_Enabled_Direct');
 		const fileUploadAllowed = RocketChat.settings.get('FileUpload_Enabled');
-
-		if (RocketChat.authz.canAccessRoom(room, user) !== true) {
+		if ((!user || (RocketChat.authz.canAccessRoom(room, user) !== true)) && (RocketChat.authz.canAccessRoom(room, file) !== true)) {
```

Comment:
I don't think it is a good idea having the same parameter receiving totally different values (in this case, first it is being passed `user` and then `file`).. I would suggest in this case to pass `file` as a third parameter.. so there wouldn't also the no need to call the same function multiple times.

also there is no need to wrap simple conditions in parenthesis like this `(RocketChat.authz.canAccessRoom(room, user) !== true)`.

Reply:
Fixed.